### PR TITLE
Bug 2054468: [Alibaba] fix deletion of more than 100 instances

### PR DESCRIPTION
**Pull request by bd233**
Because the parameter  InstanceIds of DeleteInstances and DescribeInstances support up to 100 instance IDs, an error will be reported when destroying a cluster with more than 100 nodes.

### CODE REVIEW

File: pkg/destroy/alibabacloud/alibabacloud.go [hunk 1]
```diff
@@ -1001,32 +1011,33 @@ func (o *ClusterUninstaller) deleteEcsInstances(logger logrus.FieldLogger) (err
...
 		5*time.Second,
 		5*time.Minute,
 		func() (bool, error) {
-			response, err := o.listEcsInstance(instanceIDs)
-			if err != nil {
-				return false, err
-			}
-			if response.TotalCount == 0 {
-				return true, nil
+			for _, ids := range splitArray(instanceIDs, maxInstanceCount) {
```
Comment (patrickdillon) [inline pkg/destroy/alibabacloud/alibabacloud.go:1031]: we are calling `splitArray` twice from within the same function. Here and https://github.com/openshift/installer/pull/5659/files#diff-5d31e7ab73dc252cc09331a0790fcc8b6cd3add977402effbe19a1b1b24fba39R1016. 

Can't we just call it once?

File: pkg/destroy/alibabacloud/alibabacloud.go [hunk 2]
```diff
@@ -1472,3 +1483,17 @@ func convertResourceArn(arn string) (resourceArn ResourceArn, err error) {
 	resourceArn.Arn = arn
 	return resourceArn, nil
 }
+
+func splitArray(arr []string, size int) [][]string {
+	var quantity = int(math.Ceil(float64(len(arr)) / float64(size)))
```
Comment (patrickdillon) [inline pkg/destroy/alibabacloud/alibabacloud.go:1488]: in order to improve readability, perhaps 
```suggestion
	var numRows = int(math.Ceil(float64(len(arr)) / float64(size)))
```

File: pkg/destroy/alibabacloud/alibabacloud.go [hunk 3]
```diff
@@ -1001,32 +1011,33 @@ func (o *ClusterUninstaller) deleteEcsInstances(logger logrus.FieldLogger) (err
...
-	}
-
-	logger.WithField("ecsIDs", instanceIDs).Debug("Deleting ECS instances")
-	request := ecs.CreateDeleteInstancesRequest()
-	request.InstanceId = &instanceIDs
-	request.Force = "true"
-	_, err = o.ecsClient.DeleteInstances(request)
-	if err != nil {
-		return err
+	// The paramater InstanceIds of DeleteInstances and DescribeInstances consists of up to 100 instance IDs
```
Comment (patrickdillon) [inline pkg/destroy/alibabacloud/alibabacloud.go:1014]:
```suggestion
	// The DeleteInstances and DescribeInstances throw an error if we pass more than 100 IDs, so we will split the IDs into several slices with max a length of 100.
```

**Review by patrickdillon — COMMENTED** [review]
The logic looks good. I have added some suggestions to improve readability.

**Comment by sadasu** [comment]
@bd233 Are you still interested in getting this merged?

**Comment by bd233** [comment]
> @bd233 Are you still interested in getting this merged?

Yes, is there anything I can help with?

**Comment by r4f4** [comment]
> > @bd233 Are you still interested in getting this merged?
> 
> Yes, is there anything I can help with?

Do you intend on addressing @patrickdillon's comments?

**Pull request closed without merging**